### ISSUE TITLE
False positive E3002 when using Fn to set non-primitive list property

### ISSUE THREAD

**Issue by medmunds** [awslabs/cfn-python-lint#444]
*cfn-lint version:* 0.8.3

*Description of issue:* Using a function like `!GetAtt` or `!FindInMap` to set a list property generates an erroneous E3002 "Property is an object instead of List". 

This only occurs for properties that are specced to be lists of *non*-primitive types. Properties that expect lists of primitives (e.g., List of String) seem to work fine with functions.

Minimal example (using `!GetAtt`):
```yaml
Resources:
  Custom:
    Type: AWS::CloudFormation::CustomResource
    Properties:
      ServiceToken: arn.example

  LB:
    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
    Properties:
      # SecurityGroups: {Type: List, PrimitiveItemType: String}
      # cfn-lint allows this:
      SecurityGroups: !GetAtt Custom.ExampleListOfStrings
      # LoadBalancerAttributes: {Type: List, ItemType: LoadBalancerAttribute}
      # cfn-lint rejects this with "E3002 Property is an object instead of List":
      LoadBalancerAttributes: !GetAtt Custom.ExampleListOfKeyValuePairs
```

Results: E3002 Property is an object instead of List at Resources/LB/Properties/LoadBalancerAttributes

Expected: No errors

Notes:
* The example uses a CustomResource only because that supports arbitrary GetAtts, and LoadBalancer because that has both primitive and non-primitive list properties, making it easy to demonstrate the difference in handling. But it's also reproducible using (e.g.,) `!FindInMap` to set a property to a list of objects.
* Real-world case: setting `AWS::Route53::RecordSetGroup.RecordSets` from a custom resource  https://gitlab.com/medmunds/aws-cfn-ses-domain/blob/master/example-usage.cf.yaml#L46. (This template deploys properly, but shows the cfn-lint E3002 error.)
* It looks like #43 added support for `!If` conditions in this context, but didn't cover other functions that might be used for those property values.
